### ISSUE TITLE
[apex] ApexSOQLInjection false-positive when concatenating strings

### ISSUE THREAD

**Issue by KevSlashNull** [pmd/pmd#2955]
**Affects PMD Version:** 6.21 (via [ChuckJonas/vscode-apex-pmd](https://github.com/ChuckJonas/vscode-apex-pmd/tree/master/bin/pmd)) and 6.29.0 (latest as of creating the issue)

**Description:**

If the query is not directly passed as a string literal (or multiple concatenated string literals) but instead as a variable, PMD flags the query as a rule violation when the content of the variable is concatenated.

The issue can be resolved by not passing the query as a variable to `Database.query`, however, with large queries, this makes the code harder to read (equivalent to calling a method with multiple parameters).

**Exception Stacktrace:**

No stacktrace.

**Code Sample demonstrating the issue:**

```apex
public without sharing class ExampleClass {
    public List<Account> run() {
        String query = 'SELECT Name FROM Account WHERE Active__c = true AND'
                        + ' HasSupportPlan__c = true LIMIT 5';

        return Database.query(query);
    }

    public List<Account> runWithoutRuleViolation() {
        return Database.query('SELECT Name FROM Account WHERE Active__c = true AND'
                        + ' HasSupportPlan__c = true LIMIT 5');
    }
}
```

This can also be mitigated by replacing `Database.query(query)` with `Database.query(String.escapeSingleQuotes(query))` but that’ll likely create more issues, especially when you’re not using variable binding everywhere.

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1. Create the `ExampleClass.cls` file with the content of the code sample
1. Run `pmd -d ExampleClass.cls -R rulesets/apex/quickstart.xml`
1. See that the output is the following (replace [absolute path] by the path to the `ExampleClass.cls`). The last point should not be listed because it’s just as secure as the query in `runWithoutRuleViolation`.
    ```
    /[absolute path]/ExampleClass.cls:1:     Missing ApexDoc comment
    /[absolute path]/ExampleClass.cls:2:     Missing ApexDoc comment
    /[absolute path]/ExampleClass.cls:6:     Apex classes should escape variables merged in DML query
    ```

**Running PMD through:** CLI or VS Code (Apex PMD extension)
